File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.cpp
class Solution {
public:
    int numberOfSubarrays(vector<int>& nums, int k) {
        int o = 0;
        int ans = 0;
        int prefix = 0;
        int i = 0;
        for (int j = 0; j < nums.size(); j++) {
            if (nums[j] % 2 != 0) {
                o++;
                prefix = 0;
            }
            while (o == k && i <= j) {
                if (nums[i] % 2 != 0) o--;
                i++;
                prefix++;
            }
            ans += prefix;
        }
        return ans;
    }
};

File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.py
class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        def helper(k):
            l, r = 0, 0
            curr = 0
            ans = 0

            while r < len(nums):
                curr += nums[r] % 2
                while curr > k:
                    curr -= nums[l] % 2
                    l += 1
                ans += r - l + 1
                r += 1

            return ans
        
        return helper(k) - helper(k - 1)